olive/engine/engine.py

```diff
@@ -608,7 +614,7 @@ def _get_new_model_number(self):
         while True:
             new_model_number = self._new_model_number
             self._new_model_number += 1
-            if list(self._model_cache_path.glob(f"{new_model_number}_*.json")) == []:
```

Comment:
what's the consideration to remove the file extension? would it introduce any side effect?